bingo/symbolic_regression/symbolic_regressor.py

```diff
@@ -61,94 +102,119 @@ def __init__(self, *, population_size=500, stack_size=32,
-
-        # TODO pareto front based on complexity?
-        hof = HallOfFame(5)
-
-        island = self._make_island(len(X), ea, hof)
-        self._force_diversity_in_island(island)
-
-        if self.parallel:
-            return ParallelArchipelago(island, hall_of_fame=hof)
-        else:
-            return island
-
-    def _make_island(self, dset_size, ea, hof):
-        if dset_size < 1200:
-            return Island(ea, self.generator, self.population_size, hall_of_fame=hof)
-        return FitnessPredictorIsland(ea, self.generator, self.population_size, hall_of_fame=hof,
-                                      predictor_size_ratio=800/dset_size)
-
-    def _get_clo(self, X, y, tol):
+    def _get_local_opt(self, X, y, tol):
         training_data = ExplicitTrainingData(X, y)
-        fitness = ExplicitRegression(training_data=training_data, metric=self.metric)
-        
+        fitness = ExplicitRegression(training_data=training_data,
+                                     metric=self.metric)
         optimizer = ScipyOptimizer(fitness, method=self.clo_alg, tol=tol)
         local_opt_fitness = LocalOptFitnessFunction(fitness, optimizer)
         return local_opt_fitness
 
+    def _make_island(self, dset_size, evo_alg, hof):
+        if dset_size < 1200:
+            return Island(evo_alg, self.generator, self.population_size,
+                          hall_of_fame=hof)
+        return FitnessPredictorIsland(evo_alg, self.generator,
+                                      self.population_size, hall_of_fame=hof,
+                                      predictor_size_ratio=800 / dset_size)
+
     def _force_diversity_in_island(self, island):
         diverse_pop = []
         pop_strings = set()
 
         i = 0
         while len(diverse_pop) < self.population_size:
-            i+=1
+            i += 1
             ind = self.generator()
             ind_str = str(ind)
             if ind_str not in pop_strings or i > 15 * self.population_size:
                 pop_strings.add(ind_str)
                 diverse_pop.append(ind)
-        print(f" Generating a diverse population took {i} iterations.")
         island.population = diverse_pop
 
-    def get_best_individual(self):
-        if self.best_ind is None:
-            raise ValueError("Best individual not set")
-        return self.best_ind
+    # pylint: disable=attribute-defined-outside-init
+    def _get_archipelago(self, X, y, n_processes):
+        self.component_generator = ComponentGenerator(X.shape[1])
+        for operator in self.operators:
+            self.component_generator.add_operator(operator)
+
+        self.crossover = AGraphCrossover()
+        self.mutation = AGraphMutation(self.component_generator)
+
+        self.generator = \
+            AGraphGenerator(self.stack_size, self.component_generator,
+                            use_simplification=self.use_simplification,
+                            use_python=True)
```

Comment:
Did we decide on this because of simplification?

Reply:
yep, the trampoline of python-c++-python simplification is currently very slow